src/Microsoft.Graph.Auth/ConfidentialClient/ClientCredentialProvider.cs

```diff
@@ -16,6 +16,10 @@ namespace Microsoft.Graph.Auth
     /// </summary>
     public class ClientCredentialProvider : IAuthenticationProvider
     {
+        /// <summary>
+        ///  A Scope property
+        /// </summary>
+        public string Scope { get; set; }
```

Comment:
I don't think this should be a single Scope string.  I think it should be an array of scope strings.  The goal is to allow the use of this provider for non-graph APIs.  We can't assume that non-Graph APIs only have a single scope value.